pallets/transfer-allowlist/src/lib.rs

```diff
@@ -0,0 +1,565 @@
+		AllowanceCountOverflow,
+		/// An operation expecting one or more allowances for a sending Account/Currency set, where none present
+		NoAllowancesSet,
+		/// Attempted to create allowance for existing Sending Account, Currency, and Receiver combination
+		ConflictingAllowanceSet,
+		/// CatchAll for Allowance Count arithmetic errors -- largely for coverage for errors that should be impossible
+		AllowanceCountArithmeticError,
+		/// No matching allowance for Location/Currency
+		NoMatchingAllowance,
+		/// No matching delay
+		NoMatchingDelay,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Event for successful creation of a transfer allowance
+		TransferAllowanceCreated {
+			sender_account_id: AccountIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+			receiver: Location,
+			allowed_at: BlockNumberOf<T>,
+			blocked_at: BlockNumberOf<T>,
+		},
+		/// Event for successful removal of transfer allowance perms
+		TransferAllowanceRemoved {
+			sender_account_id: AccountIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+			receiver: Location,
+			allowed_at: BlockNumberOf<T>,
+			blocked_at: BlockNumberOf<T>,
+		},
+		/// Event for successful removal of transfer allowance perms
+		TransferAllowancePurged {
+			sender_account_id: AccountIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+			receiver: Location,
+		},
+		/// Event for Allowance delay update
+		TransferAllowanceDelaySet {
+			sender_account_id: AccountIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+			delay: BlockNumberOf<T>,
+		},
+		/// Event for Allowance delay removal
+		TransferAllowanceDelayRemoval {
+			sender_account_id: AccountIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+		},
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T>
+	where
+		<T as frame_system::Config>::AccountId: Into<Location>,
+	{
+		#[transactional]
+		#[pallet::call_index(0)]
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(2, 2).ref_time())]
+		/// Adds a transfer allowance for a sending Account/Currency.
+		/// Allowance either starts at the current block + the delay set for the account,
+		/// if a delay is present.
+		/// or block 0 if no delay is present.
+		/// Important! Account/Currency sets with an allowance set are restricted to just the allowances added for the account -
+		/// to have unrestricted transfers allowed for the sending Account and Currency, no allowances should be present.
+		///
+		/// Running this for an existing allowance generates a new allowance based on the current delay, or lack thereof
+		pub fn add_transfer_allowance(
+			origin: OriginFor<T>,
+			currency_id: CurrencyIdOf<T>,
+			receiver: Location,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+
+			let allowance_details = match Self::sender_currency_delay(&account_id, currency_id) {
+				Some(delay) => AllowanceDetails {
+					allowed_at: <frame_system::Pallet<T>>::block_number().saturating_add(delay),
```

Comment:
I would increase with `check_add` or `ensure_add` here. If we have an error obtaining a huge `delay` it silently works when it should not

Reply:
My thinking was that if the delay + the current bock is greater than the max number of blocks then it would make sense to just cap at the max # of blocks.  But returning an error would also make sense, I'm good with changing if we want that changed.

Reply:
Ok, I see what you mean. Then it's ok for me as it is.